lib/Model/EventTree.js

```diff
@@ -0,0 +1,114 @@
+module.exports = EventTree;
+
+function EventTree(parent, segment) {
+  this.parent = parent;
+  this.segment = segment;
+  this.children = null;
+  this.listeners = null;
+}
+
+EventTree.prototype.destroy = function() {
+  // Ignore calls to destroy a root node
+  if (!this.parent) return;
+
+  // Remove reference this node from its parent
+  if (hasOtherKeys(this.parent.children, this.segment)) {
+    delete this.parent.children[this.segment];
+    return;
+  }
+  this.parent.children = null;
+
+  // Destroy parent if it no longer has any dependents
+  if (!this.parent.listeners) {
+    this.parent.destroy();
+  }
+};
+
+EventTree.prototype.getChild = function(segments) {
```

Comment:
`getChild` and `getOrCreateChild` could be underscore-prefixed to indicate they're only used from within this file.